vivo-project/Vitro-angular
# [Issue 6] Dynamic API feature store

**[Vitro-angular GitHub issue](https://github.com/vivo-project/Vitro-angular/issues)**: https://github.com/vivo-project/Vitro-angular/issues/7

# What does this pull request do?
Adds feature store that fetches dynamic API REST Docs on initialization.

# What's new?
REST Docs feature store with state, reducer, effects, and selectors.

# How should this be tested?
1. start up vitro with CORS filter and dynamic action n files
2. `npm run start`
3. http://localhost:4200
4. open developer tools
5. should see console logs with rest docs as json

> anticipating vitro http://localhost:8080/api by default

![image](https://user-images.githubusercontent.com/8352733/172927773-ae135baa-11cd-4ed2-b9b8-e2da2c910ba0.png)

![image](https://user-images.githubusercontent.com/8352733/172928194-bf245720-d00c-425c-8276-6283c042c76c.png)

## Checklist
_This checklist provides a reminder of what we are going to look for when reviewing your PR. You need not complete this checklist prior to creating your PR (draft PRs are always welcome). If you are unsure about an item in the checklist, don't hesitate to ask. We're here to help!_

- [x] If my PR includes new, third-party dependencies (in `package.json`), I've made sure their licenses align with the [Vitro-angular License](https://github.com/vivo-project/Vitro-angular/blob/main/LICENSE).
